osu.Game.Rulesets.Osu/Skinning/Legacy/LegacyReverseArrow.cs

```diff
@@ -68,11 +71,42 @@ private void onHitObjectApplied(DrawableHitObject drawableObject)
                                                     .OverlayElementContainer.Add(proxy);
         }
 
+        private void updateStateTransforms(DrawableHitObject hitObject, ArmedState state)
+        {
+            const double move_out_duration = 35;
+            const double move_in_duration = 250;
+            const double total = 300;
+
+            switch (state)
+            {
+                case ArmedState.Idle:
+                    if (shouldRotate)
+                    {
+                        InternalChild.ScaleTo(1.3f, move_out_duration)
+                                     .RotateTo(5.625f)
+                                     .Then()
+                                     .ScaleTo(1f, move_in_duration)
+                                     .RotateTo(-5.625f, move_in_duration)
+                                     .Loop(total - (move_in_duration + move_out_duration));
+                    }
+                    else
+                    {
+                        InternalChild.ScaleTo(1.3f)
+                                     .Then()
+                                     .ScaleTo(1f, move_in_duration)
+                                     .Loop(total - (move_in_duration + move_out_duration));
+                    }
```

Comment:
Stable, for reference, as far as I can tell:

```csharp
            if (reverse)
            {
                if (SkinManager.UseNewLayout)
                {
                    SpriteHitCircleText.Rotation = angle;
                    for (int i = appearTime; i < startTime; i += 300)
                    {
                        int length = Math.Min(300, startTime - i);
                        SpriteHitCircleText.Transformations.Add(new Transformation(TransformationType.Scale, 1.3f, 1, i, i + length, EasingTypes.Out));
                    }
                }
                else
                {
                    for (int i = appearTime; i < startTime; i += 300)
                    {
                        int length = Math.Min(300, startTime - i);
                        SpriteHitCircleText.Transformations.Add(new Transformation(TransformationType.Scale, 1.3F, 1, i, i + length));
                        SpriteHitCircleText.Transformations.Add(new Transformation(TransformationType.Rotation, angle + OsuMathHelper.Pi / 32, angle - OsuMathHelper.Pi / 32, i, i + length));
                    }
                }
            }
        }
```

(https://github.com/peppy/osu-stable-reference/blob/3ea48705eb67172c430371dcfc8a16a002ed0d3d/osu!/GameplayElements/HitObjects/Osu/HitCircleSliderEnd.cs#L77-L98).

From a surface-level look:

- Missing `Out` easing in the `UseNewLayout` (non-v1.0) path
- Durations are slightly different (stable doesn't do that many easings with `move_{in,out}_duration`, it just shoves in 300ms constant)
- Stable has a special case that ensures the last loop is shortened if there's not enough time to play it fully out otherwise (see `length` treatment)

Should I treat these implementation differences as intentional for now?

Reply:
I've made some changes here, which I thought I already did... may have accidentally lost the changes.